src/common/ddax_deep_flush.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2017, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * ddaxe_deep_flush.c -- Internal utility functions for flushing
+ * a memory range residing on a DAX device.
+ * Currently only used on Linux.
+ */
+
+#define _GNU_SOURCE
+
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "os.h"
+#include "file.h"
+#include "out.h"
+#include "ddax_deep_flush.h"
+#include "libpmem.h"
+
+#define DAX_REGION_ID_LEN 4
+#define MAX_UNIQUE_REGIONS 1024
+
+/*
+ * ddax_region_find - looks for region number for specified dev_id
+ */
+int
+ddax_region_find(os_dev_t dev_id)
+{
+
+	LOG(2, "ddax_region_find %lu", (unsigned long)dev_id);
+#ifndef _WIN32
```

Comment:
This entire file makes no sense on windows - it will have to be implemented differently.
Please abstract away platform specific code in platform specific files. This allows you to entirely eliminate `#ifdef PLATFORM` because you can control what is executed through changing what file is compiled.
Also, this will break FreeBSD - all the more argument on changing this...